### PR TITLE
[BUG]: Historical empty bug

### PR DESCRIPTION
**Pull request by shivani170**
# Description

Fixed when Deployment History > Configuration Page is breaking due to an unhandled null result when there is no collected history.

Fixes https://github.com/devtron-labs/devtron/issues/1393

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Tested Out with no history in deployment Configuration
- [x] Tested Out with history of  deployment Configuration

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/app/details/cdDetails/CDDetails.tsx [hunk 1]
```diff
@@ -143,18 +144,25 @@ export default function CDDetails(){
...
         } catch (err) {
             showError(err);
+            setLoader(false);
         }
-    }
-    }, [pipelineId]);
+
+        return (): void => {
+            if (showTemplate) {
+                setShowTemplate(false);
```
Comment (sohel-rp) [inline src/components/app/details/cdDetails/CDDetails.tsx:163]: Remove this as we don't need this in CDDetails.

File: src/components/app/details/cdDetails/DeploymentHistoryConfigTabView.tsx [hunk 2]
```diff
@@ -1,87 +1,70 @@
...
-            });
-
-            if (!showTemplate) {
-                setShowTemplate(true);
-            }
-        } catch (err) {
-            showError(err);
-            setLoader(false);
+        if (!showTemplate) {
+            setShowTemplate(true)
```
Comment (sohel-rp) [inline src/components/app/details/cdDetails/DeploymentHistoryConfigTabView.tsx:65]: Suggestion - we can set showTemplate as false on unmount if not doing explicitly.

File: src/components/app/details/cdDetails/CDDetails.tsx (hunk 1 above)
Comment (shivani170) [reply src/components/app/details/cdDetails/CDDetails.tsx:163]: done

File: src/components/app/details/cdDetails/DeploymentTemplateWrapper.tsx [hunk 3]
```diff
@@ -15,12 +15,12 @@ interface TemplateConfiguration {
 function DeploymentTemplateWrapper({ setShowTemplate, deploymentTemplatesConfiguration, setBaseTimeStamp, baseTimeStamp }: TemplateConfiguration) {
     const match = useRouteMatch();
     const { triggerId } = useParams<{ triggerId: string }>();
-    const deploymentTemplateFilteredTrigger = deploymentTemplatesConfiguration.find(
+    const deploymentTemplateFilteredTrigger =  deploymentTemplatesConfiguration?.find(
         (dt) => dt.wfrId.toString() === triggerId,
     );
-    const isLastDeploymentTemplatesConfiguration =  deploymentTemplatesConfiguration[deploymentTemplatesConfiguration.length - 1].wfrId.toString() === triggerId
+    const isLastDeploymentTemplatesConfiguration =  deploymentTemplatesConfiguration && deploymentTemplatesConfiguration[deploymentTemplatesConfiguration.length - 1]?.wfrId.toString() === triggerId
```
[the remark this answers is not in the document]
Comment (shivani170) [reply src/components/app/details/cdDetails/DeploymentTemplateWrapper.tsx]: fixed

**Review by vivek-devtron — APPROVED** [review]

**Review by sohel-rp — APPROVED** [review]

**Review by pghildiyal — APPROVED** [review]

**Pull request merged**